### PR TITLE
WIP: App-type fixtures

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal

In order to facilitate investigating the setting of `app_type` in different frameworks, added fixtures for Rake, Rack, Mailman, and Que.

Further fixtures to be added include Resque, Shoryuken, and in-rails-app Resque, Que, Delayed-job, Rake, & Mailman

### CODE REVIEW

File: features/fixtures/docker-compose.yml [hunk 1]
```diff
@@ -272,3 +277,51 @@ services:
...
+      args:
+        - RUBY_VERSION
+    environment:
+      - BUGSNAG_API_KEY
+      - BUGSNAG_ENDPOINT
+    ports:
+      - "3008:8080"
+    restart: 'no'
+
+  rake:
```
Comment (snmaynard) [inline features/fixtures/docker-compose.yml]: Could we not add a rake task in a rails app?

File: features/fixtures/mailman/app/app.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require 'mailman'
+require 'bugsnag'
+require 'pp'
```
Comment (snmaynard) [inline features/fixtures/mailman/app/app.rb]: pp is unused

File: features/fixtures/que/.dockerignore [hunk 3]
```diff
@@ -0,0 +1 @@
+app/Gemfile.lock
```
Comment (snmaynard) [inline features/fixtures/que/.dockerignore]: how comes?

File: features/fixtures/mailman/app/emails/Unhandled.eml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+MIME-Version: 1.0
```
Comment (snmaynard) [inline features/fixtures/mailman/app/emails/Unhandled.eml]: filename should be all lowercase?

File: features/fixtures/mailman/app/emails/Handled.eml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+MIME-Version: 1.0
```
Comment (snmaynard) [inline features/fixtures/mailman/app/emails/Handled.eml]: filename should be all lowercase?

File: features/fixtures/mailman/app/run.sh [hunk 6]
```diff
@@ -0,0 +1 @@
+cat target.eml | bundle exec ruby app.rb
```
Comment (snmaynard) [inline features/fixtures/mailman/app/run.sh]: what is `target.eml`?

File: features/fixtures/que/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,14 @@
+ARG RUBY_VERSION
+FROM ruby:$RUBY_VERSION
+
+ARG RUBY_VERSION
```
Comment (snmaynard) [inline features/fixtures/que/Dockerfile]: Why is this in here twice?

File: features/fixtures/rack/.dockerignore [hunk 8]
```diff
@@ -0,0 +1 @@
+app/Gemfile.lock
```
Comment (snmaynard) [inline features/fixtures/rack/.dockerignore]: how comes

File: features/fixtures/rack/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,16 @@
+ARG RUBY_VERSION
+FROM ruby:$RUBY_VERSION
+
+ARG RUBY_VERSION
```
Comment (snmaynard) [inline features/fixtures/rack/Dockerfile]: In here twice - why?

File: features/fixtures/mailman/.dockerignore [hunk 10]
```diff
@@ -0,0 +1,2 @@
+Gemfile.lock
```
[the remark this answers is not in the document]
Comment (Cawllec) [reply features/fixtures/mailman/.dockerignore]: I've previously ignored lockfiles as the dependencies tested against should be flexible based on minor/patch versions.  The only reason there would ever be a lockfile here would be if fixtures were being run manually.

[the remark this answers is not in the document]
Comment (snmaynard) [reply features/fixtures/mailman/.dockerignore]: Add a comment saying this in the relevant places. Ignoring lock files is definitely commentworthy.

File: features/sidekiq.feature [hunk 11]
```diff
@@ -65,9 +64,8 @@ Scenario Outline: A handled RuntimeError can be notified
   And the event "context" equals "HandledError@default"
   And the event "severity" equals "warning"
   And the event "severityReason.type" equals "handledException"
+  And the event "app.type" equals "sidekiq"
   And the exception "errorClass" equals "RuntimeError"
-  And the "file" of stack frame 0 equals "/usr/src/app.rb"
```
Comment (snmaynard) [inline features/sidekiq.feature]: Why have these been removed?

**Comment by imjoehaines** [comment]
Continued in https://github.com/bugsnag/bugsnag-ruby/pull/616

**Pull request closed without merging**